### PR TITLE
Build

### PR DESCRIPTION
**Pull request by jburel**
Build ol3-viewer.js as part of the build system
remove the compiled code from repository

the production command is tested on cowfish
to test locally: ``npm run dev`` and ``npm run debug``

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [iviewer-merge#21](https://ci.openmicroscopy.org/job/iviewer-merge/21/). See the [console output](https://ci.openmicroscopy.org/job/iviewer-merge/21/consoleText) for more details.
Possible conflicts:
  - Conflict detection failed
    - libs/ol3-viewer.js

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [iviewer-merge#22](https://ci.openmicroscopy.org/job/iviewer-merge/22/). See the [console output](https://ci.openmicroscopy.org/job/iviewer-merge/22/consoleText) for more details.
Possible conflicts:
  - Conflict detection failed
    - libs/ol3-viewer.js

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [iviewer-merge#23](https://ci.openmicroscopy.org/job/iviewer-merge/23/). See the [console output](https://ci.openmicroscopy.org/job/iviewer-merge/23/consoleText) for more details.
Possible conflicts:
  - Conflict detection failed
    - libs/ol3-viewer.js

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [iviewer-merge#24](https://ci.openmicroscopy.org/job/iviewer-merge/24/). See the [console output](https://ci.openmicroscopy.org/job/iviewer-merge/24/consoleText) for more details.
Possible conflicts:
  - Conflict detection failed
    - libs/ol3-viewer.js

**Comment by jburel** [comment]
--exclude

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/view/Web/job/WEB-DEV-trial-deploy/131/ is green

**Comment by waxenegger** [comment]
As discussed, in regards to the 2 DEV configurations which do not need a rebuild of ol3-viewer.js on every call:

`#!/bin/bash

# we remove and compile ol3-viewer.js in the following cases:
# 1. we are not in DEV mode
# 2. we are in DEV mode and the file does not exist
# 3. we are in DEV mode, the file exists but is older than 24 hours
# Note: 'default' is to compile
COMPILE_OL3_VIEWER=0
if [ "$#" -gt 0 ] && [ "$1" = "DEV" ]; then
    if [ -f "libs/ol3-viewer.js" ]; then
        TIMESTAMP=`date +'%s'`
        TIMESTAMP_OL3_VIEWER_JS=`stat -c %Y libs/ol3-viewer.js`
        TIMESTAMP_DIFF=$(($TIMESTAMP - $TIMESTAMP_OL3_VIEWER_JS))
        if [ $TIMESTAMP_DIFF -lt $((60 * 60 * 24)) ]; then
            COMPILE_OL3_VIEWER=1
        fi
    fi
fi

if [ $COMPILE_OL3_VIEWER -eq 0 ]; then
    echo "preparing ol3-viewer"
    rm  -rf libs
    mkdir -p libs
    ant
else
    echo "omitting ol3-viewer build"
fi
`

File: ol3_viewer_prepare.sh [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
-echo "preparing ol3-viewer"
-rm  -rf libs
-mkdir -p libs
-ant
+compile_viewer=0
+if [ "$#" -gt 0 ] && [ "$1" = "DEV" ]; then
+    if [ -f "libs/ol3-viewer.js" ]; then
+        timestamp=$(date +%s)
+        echo $timestamp
+        timestamp_viewer=$(stat -f%c libs/ol3-viewer.js)
```
Comment (waxenegger) [inline ol3_viewer_prepare.sh]: stat -f%c libs/ol3-viewer.js would not work  I'm pretty sure

Comment (jburel) [inline ol3_viewer_prepare.sh]: stat -c does not work on mac
looking into alternative

Comment (waxenegger) [inline ol3_viewer_prepare.sh]: also I think that the echo $timestamp slipped in there for debug info...

is it a problem to stick to upper case variables for bash scripts. it seems to be kind of a style guide for shell scripts (as far as I have experienced it)

Comment (jburel) [inline ol3_viewer_prepare.sh]: the echo should not be there!

Comment (jburel) [inline ol3_viewer_prepare.sh]: After discussion with @waxenegger, 
if the file is present in "dev" mode, we won't check the time.
Will add a note to the readme (other PR already opened)

**Comment by waxenegger** [comment]
Sorry, the timestamp logic which would have forced a compile for development modes is kind of unnecessary, on second thought. 

If ol3-code has been touched it would have to be compiled as well (regardless of any time set) which is mentioned in the README anyhow, and otherwise there is no need to force it to begin with.

**Comment by waxenegger** [comment]
@jburel Tested all 3 scenarios and it behaves as expected.

**Comment by jburel** [comment]
Note that the compile code ``libs/ol3-viewer.js`` has been removed in https://github.com/ome/omero-iviewer/pull/11

**Pull request merged**